tests/torch/test_sota_checkpoints.py

```diff
@@ -9,684 +9,535 @@
 from tests.shared.paths import TEST_ROOT
+from tests.torch.helpers import Command
 
-BG_COLOR_GREEN_HEX = "ccffcc"
-BG_COLOR_YELLOW_HEX = "ffffcc"
-BG_COLOR_RED_HEX = "ffcccc"
+DIFF_TARGET_PT_MIN = -0.1
+DIFF_TARGET_PT_MAX = 0.1
+DIFF_TARGET_OV_MIN = -0.01
+DIFF_TARGET_OV_MAX = 0.01
+PYTORCH = "PT"
+OPENVINO = "OV"
+TRAIN = "TRAIN"
 
 
+@dataclass
 class EvalRunParamsStruct:
-    def __init__(
-        self,
-        config_name_: str,
-        reference_: Optional[str],
-        expected_: float,
-        metric_type_: str,
-        dataset_name_: str,
-        sample_type_: str,
-        resume_file_: str,
-        batch_: int,
-        mean_val_: Optional[str],
-        scale_val_: Optional[str],
-        diff_fp32_min_: float,
-        diff_fp32_max_: float,
-        model_name_: str,
-        diff_target_min_: float,
-        diff_target_max_: float,
-        multiprocessing_distributed: bool,
-    ):
-        self.config_name_ = config_name_
-        self.reference_ = reference_
-        self.expected_ = expected_
-        self.metric_type_ = metric_type_
-        self.dataset_name_ = dataset_name_
-        self.sample_type_ = sample_type_
-        self.resume_file_ = resume_file_
-        self.batch_ = batch_
-        self.mean_val_ = mean_val_
-        self.scale_val_ = scale_val_
-        self.diff_fp32_min_ = diff_fp32_min_
-        self.diff_fp32_max_ = diff_fp32_max_
-        self.model_name_ = model_name_
-        self.diff_target_min_ = diff_target_min_
-        self.diff_target_max_ = diff_target_max_
-        self.multiprocessing_distributed = multiprocessing_distributed
+    """
+    Contain data about quantization of the model.
+    """
+
+    config_name: str
+    reference: Optional[str]
+    target_ov: float
+    target_pt: float
+    metric_type: str
+    dataset_name: str
+    sample_type: str
+    resume_file: str
+    batch: int
+    diff_fp32_min: float
+    diff_fp32_max: float
+    model_name: str
+    diff_target_ov_min: float
+    diff_target_ov_max: float
+    diff_target_pt_min: float
+    diff_target_pt_max: float
+    multiprocessing_distributed: bool
+    skip_ov: str
+
+
+@dataclass
+class ResultInfo:
+    """
+    Contain data about result of test.
+    """
+
+    model_name: str
+    backend: str
+    metric_type: Optional[str] = None
+    expected: Optional[float] = None
+    measured: Optional[float] = None
+    diff_fp32: Optional[float] = None
+    diff_target: Optional[float] = None
+    status: Optional[str] = None
+
+    def to_dict(self):
+        return {
+            "Model": self.model_name,
+            "Backend": self.backend,
+            "Metrics type": self.metric_type,
+            "Expected": self.expected,
+            "Measured": self.measured,
+            "Diff FP32": self.diff_fp32,
+            "Diff target": self.diff_target,
+            "Status": self.status,
+            "Build url": os.environ.get("BUILD_URL", ""),
+        }
+
+
+TEST_RESULT: List[ResultInfo] = []
```

Comment:
Perhaps this could also be encapsulated within a class-level fixture instead of being a global var.

Reply:
There is any problem to use with global var for this case?

Reply:
Global state is bad, period, if it can be avoided. Specifically in the situation of our E2E tests we are launching these in a separate python process. As long as it is true, I suppose the global vars can functionally be used and do not pose visible problems. However, if this situation changes in the future, e.g. we will be launching the E2E tests along with some other tests, or add a test class to the E2E testing suite that would try to reuse this global state, we might be ending with hard-to-discover subtle bugs.

Your state actually belongs to the E2E testing specifically. If you encapsulate E2E test code in a speicific class (it seems that this is done already), then this state should also be associated with the class. With fixtures it seems to be done using a class-scoped fixture that would be returning an object reference that is shared across all tests in a class, and you can use this object to store test results as tests execute. See https://stackoverflow.com/questions/28288385/pytest-how-to-pass-an-argument-to-setup-class again - if I'm not mistaken, the `A_helper` object will only be created once per class, accessible in every test case and can therefore serve as a storage for class-scope state such as the list of the test results.

Reply:
` Specifically in the situation of our E2E tests we are launching these in a separate python process. As long as it is true, I suppose the global vars can functionally be used and do not pose visible problems. ` 

Ok i will keep as it.

Reply:
> `Specifically in the situation of our E2E tests we are launching these in a separate python process. As long as it is true, I suppose the global vars can functionally be used and do not pose visible problems.`
> 
> Ok i will keep as it.

As a good developer, you should be preparing for the future, and implementing best practices in code even if they don't seem to have an immediate effect, especially if you are pointed towards them explicitly. So I, as a maintainer, want you to do so and won't approve the PR otherwise.

Reply:
Ok, please look on 
https://github.com/openvinotoolkit/nncf/blob/6130fca6dc199ea0b0a8a4dfa785e0da9217ca66/tests/torch/test_sota_checkpoints.py#L257-L270

Is it correct way to use fixture to collect data?

Reply:
This is how it's supposed to be done, yes. Details could be adjusted if necessary. It worked well to date, anyway.